src/libraries/System.Net.Security/src/System/Net/CertificateValidationPal.Windows.cs

```diff
@@ -89,6 +89,28 @@ internal static partial class CertificateValidationPal
             return result;
         }
 
+        // Check that local certificate was used by schannel.
+        internal static bool IsLocalCertificateUsed(SafeDeleteContext securityContext)
+        {
+            SafeFreeCertContext? localContext = null;
+            try
+            {
+                if (SSPIWrapper.QueryContextAttributes_SECPKG_ATTR_LOCAL_CERT_CONTEXT(GlobalSSPI.SSPISecureChannel, securityContext, out localContext) &&
+                    localContext != null)
+                {
+                    return !localContext.IsInvalid;
+                }
+            }
+            finally
+            {
+                localContext?.Dispose();
+            }
+
+            // Some older Windows do not support that. This is only called when client certificate was provided
```

Comment:
Do unsupported platforms throw in this case? If so, won't that cause a perf regression?

Reply:
no, I don't think it throws. The Windows7 fails to even pass the first `IsMutuallyAuthenticate` Assert. The explanation is in test above and that test is also disabled on Windows7.  So I also disabled new test on Win7 (and it is about of support anyway)